### PR TITLE
reopen access/error logs on sighup

### PR DESCRIPTION
**Pull request by kazegusuri**
by using [github.com/client9/reopen](https://github.com/client9/reopen)

### CODE REVIEW

File: cmd/gaurun/gaurun.go [hunk 1]
```diff
@@ -89,6 +92,26 @@ func main() {
...
+					if err := accessLogReopener.Reopen(); err != nil {
+						gaurun.LogError.Warn(fmt.Sprintf("failed to reopen access log: %v", err))
+					}
+					if err := errorLogReopener.Reopen(); err != nil {
+						gaurun.LogError.Warn(fmt.Sprintf("failed to reopen error log: %v", err))
+					}
+				}
+			}
+		}
+	}()
```
Comment (cubicdaiya) [inline cmd/gaurun/gaurun.go]: Looks too deep nest to me.

**Comment by kazegusuri** [comment]
@cubicdaiya refactored signal handler to avoid deep nest.

File: gaurun/log.go [hunk 2]
```diff
@@ -63,10 +67,11 @@ func InitLog(outString string, levelString string) (zap.Logger, error) {
 
 	var level zap.Level
 	if err := level.UnmarshalText([]byte(levelString)); err != nil {
-		return nil, err
+		return nil, nil, err
 	}
 
-	return zap.New(encoder, level, zap.Output(writer), zap.ErrorOutput(writer)), nil
+	writeSyncer := zap.AddSync(writer)
```
Comment (cubicdaiya) [inline gaurun/log.go]: Why is zap.AddSync added?

File: cmd/gaurun/gaurun.go [hunk 3]
```diff
@@ -82,6 +85,20 @@ func main() {
 		}
 	}
 
+	sigChan := make(chan os.Signal, 1)
+	signal.Notify(sigChan, syscall.SIGHUP)
+
+	sighupHandler := func() {
```
Comment (cubicdaiya) [inline cmd/gaurun/gaurun.go]: This function should be defined as normal function.

Comment (cubicdaiya) [inline cmd/gaurun/gaurun.go]: Oh, sorry. I thought good.

File: gaurun/log.go (hunk 2 above)
Comment (kazegusuri) [inline gaurun/log.go]: zap.Output requires zap.WriteSyncer. io.Writer needs to use zap.AddSyncer to implement zap.WriteSyncer.

**Pull request merged**